### PR TITLE
Switching to a population level object

### PR DESCRIPTION
**Pull request by LeahPrice**
Instead of having a particle level object, this version uses a population level object with values stored in an std::vector and log weights stored in an arma::vec.

An MCMC step was also added to Initialise().

### CODE REVIEW

**Comment by LeahPrice** [comment]
Sure, thank you for the lines to fix this! It looks like it passed this time.

**Comment by eddelbuettel** [comment]
I think with that we're good.  Let's wait for @adamjohansen to take a peek at it, but I think we got this one covered now.  Yay!

(As for going forward:  go to the travis-ci.org website and assign _your_ rcppsmc (and/or other repos you want tested.  You need to turn this one at a time for new repos.  Forks usually work as is, but because you were not set up before it did not automagically work.   This is hugely worth it so I recommend you look into it.)

**Comment by LeahPrice** [comment]
Cool, thank you! Yeah it'll be nice to not get a surprise failure with Travis CI when I do future pull requests.

**Comment by eddelbuettel** [comment]
Thanks for the additional commits.  I think it might be best to just hold off committing until @adamjohansen is back online (which should be today) and we can finalize this.

File: inst/include/sampler.h [hunk 1]
```diff
@@ -112,6 +115,8 @@ namespace smc {
         double GetParticleLogWeightN(long n) const { return pPopulation.GetLogWeightN(n); }
         ///Return the unnormalized weight of particle n
         double GetParticleWeightN(long n) const { return pPopulation.GetWeightN(n); }
+        ///Return the unnormalized weight of particle n
```
Comment (adamjohansen) [inline inst/include/sampler.h]: Does this comment match what the function actually does?

Comment (LeahPrice) [inline inst/include/sampler.h]: Ah yes you're right, thanks. I meant to say "Return the unnormalised weights of the particles". I'll fix that up now.

**Comment by LeahPrice** [comment]
Also sorry Adam, I for some reason thought that my commits today wouldn't show up on this pull request. I hope this wasn't too annoying for you to review.

**Comment by eddelbuettel** [comment]
>  I for some reason thought that my commits today wouldn't show up on this pull request

You could revert these commits. Not yet saying you should, just saying you could.

If you want commits to be somewhere else, create a branch first.  What happened here is perfectly logical and a feature -- one wants to be able to add to / augment a  pull request.

**Comment by eddelbuettel** [comment]
But I think we're good. here.  Let me merge this now, and then we can do smaller / simpler follow-up PRs.

**Comment by eddelbuettel** [comment]
@LeahPrice One thing you want to look into is defining your username and email:

```
*   134a9cf - (HEAD -> master, origin/master, origin/HEAD) Merge pull request #2 from LeahPrice/PopulationLevel (36 seconds ago) <Dirk Eddelbuettel>
|\
| * 43f3a63 - Rcpp attributes and example namespaces (8 hours ago) <Unknown>
| * 8ef8afd - Updating copyright headers (8 hours ago) <Unknown>
| * 7eca96b - Adding RcppArmadillo to installs (2 days ago) <Unknown>
| * 834d3c7 - Switching to a population level object (2 days ago) <Unknown>
|/
* 63bb33d - small update for registration (4 days ago) <Dirk Eddelbuettel>
```

"Unknown" is not the type of credit you want in the long run.

**Comment by LeahPrice** [comment]
Okay cool, thanks. I still haven't fixed that comment for the function so I'll do that in another pull request then.

**Pull request merged**